react-electron-poc/src/renderer/components/layout/Layout.tsx

```diff
@@ -100,134 +242,212 @@ const Layout = () => {
             );
+
+            if (panelManager.current.panelsInfo.size >= startingTabs) return;
+
             const ohebPanel = panelManager.current.addPanel(
                 'ScriptureTextPanelHtml',
                 {
                     shortName: 'OHEB',
                     editable: false,
                     ...scrRef,
                     updateScrRef,
+                    useVirtualization,
                     browseBook,
-                } as ScriptureTextPanelHtmlProps,
+                } as ScriptureTextPanelStringProps,
                 {
                     position: {
                         direction: 'right',
                         referencePanel: csbPanel.id,
                     },
                 },
             );
+
+            if (panelManager.current.panelsInfo.size >= startingTabs) return;
+
             panelManager.current.addPanel(
                 'ScriptureTextPanelSlate',
                 {
                     shortName: 'zzz6',
                     editable: true,
                     ...scrRef,
                     updateScrRef,
+                    useVirtualization,
                     browseBook,
-                } as ScriptureTextPanelHtmlProps,
+                } as ScriptureTextPanelStringProps,
                 {
                     position: {
                         direction: 'below',
                         referencePanel: csbPanel.id,
                     },
                 },
             );
+
+            if (panelManager.current.panelsInfo.size >= startingTabs) return;
+
             panelManager.current.addPanel(
                 'ScriptureTextPanelHtml',
                 {
                     shortName: 'NIV84',
                     editable: false,
                     ...scrRef,
                     updateScrRef,
+                    useVirtualization,
                     browseBook,
-                } as ScriptureTextPanelHtmlProps,
+                } as ScriptureTextPanelStringProps,
                 {
                     position: {
                         direction: 'below',
                         referencePanel: ohebPanel.id,
                     },
                 },
             );
+
+            if (panelManager.current.panelsInfo.size >= startingTabs) return;
+
             const zzz1Panel = panelManager.current.addPanel(
                 'ScriptureTextPanelHtml',
                 {
                     shortName: 'zzz1',
                     editable: true,
                     ...scrRef,
                     updateScrRef,
+                    useVirtualization,
                     browseBook,
-                } as ScriptureTextPanelHtmlProps,
+                } as ScriptureTextPanelStringProps,
                 {
                     position: {
                         direction: 'below',
                         referencePanel: ohebPanel.id,
                     },
                 },
             );
+
+            /* if (startingTabs < 6) return;
```

Comment:
Is this comment still needed?

Reply:
Good call. I will take it out :) thanks!